### PR TITLE
Fix Vendor Affirmation for SLED

### PR DESCRIPTION
**Pull request by volodymyrkatkalov**
- Related ticket: https://progress.opensuse.org/issues/167332
- Verification run: https://openqa.suse.de/tests/overview?distri=sle&version=15-SP6&build=20241222-1&groupid=594

### CODE REVIEW

File: lib/security/openssl_misc_utils.pm [hunk 1]
```diff
@@ -55,7 +55,8 @@ sub install_openssl {
             install_11_workaround_when_no_legacy();
             return;
         }
-        add_suseconnect_product('sle-module-legacy');
+        record_info('Extensions List', script_output("SUSEConnect --list-extensions"));
+        add_suseconnect_product('sle-module-legacy') unless is_sled;
```
Comment (dzedro) [inline lib/security/openssl_misc_utils.pm:59]: This was broken also before, but this is redundant of `has_no_legacy_module`
IMO there should be `else` to empty line 53

File: lib/version_utils.pm [hunk 2]
```diff
@@ -405,6 +406,24 @@ sub is_sle {
...
+Check if distribution is SLED with optional filter for:
+Version: <=12-sp3 =12-sp1 >11-sp1 >=15 15+ (>=15 and 15+ are equivalent)
+=cut
+
+sub is_sled {
+    my $query = shift;
+    my $version = shift // get_var('VERSION');
+
+    # Check if it is SLE and specifically SLED
+    return 0 unless is_sle($query, $version);
```
Comment (dzedro) [inline lib/version_utils.pm:420]: Is this about is_sled or SLE version ? You just copied is_sle description, that does have  nothing with `is_sled` you will not even compare the SLED version at the end.

Why complicate simple thing ? Twisting it twice.
Why not just return 1 if ...SLED...

Comment (volodymyrkatkalov) [reply lib/version_utils.pm]: Reduced to just:
`=head2 is_sled

Check if distribution is SLED
=cut

sub is_sled {
    return 0 unless is_sle();
    return 0 unless check_var('SLE_PRODUCT', 'sled') || get_var('FLAVOR') =~ /-Desktop/i;
    return 1;
}`

Comment (dzedro) [reply lib/version_utils.pm]: `return 0 unless is_sle();` makes this broken, SLED is also SLE so also SLES would pass here.

File: lib/security/openssl_misc_utils.pm [hunk 3]
```diff
@@ -53,10 +53,12 @@ sub install_openssl {
...
             install_11_workaround_when_no_legacy();
-            return;
+        } else {
+            record_info('Extensions List', script_output("SUSEConnect --list-extensions"));
+            add_suseconnect_product('sle-module-legacy');
+            zypper_call 'in openssl-1_1';
         }
-        add_suseconnect_product('sle-module-legacy');
-        zypper_call 'in openssl-1_1';
+        return;
```
Comment (dzedro) [inline lib/security/openssl_misc_utils.pm:61]: return is not needed with else

File: lib/version_utils.pm (hunk 2 above)
Comment (volodymyrkatkalov) [reply lib/version_utils.pm]: @dzedro What would be a better way? sles passes, then SLE_PRODUCT check is done

Comment (dzedro) [reply lib/version_utils.pm]: 1. Check how return works
2. [Why not just return 1 if ...SLED...](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20845#discussion_r1891640788)

Comment (volodymyrkatkalov) [reply lib/version_utils.pm]: @dzedro  Thanks for your input. The is_sle() check is included to account for the || get_var('FLAVOR') =~ /-Desktop/i part, ensuring cases are handled where FLAVOR might suggest a desktop distribution even if SLE_PRODUCT isn’t explicitly set to 'sled'. If it's absolutely guaranteed that all SLED systems will always have SLE_PRODUCT set to 'sled', we could streamline this further. However, I haven't seen confirmation of that assumption. 

While I can rewrite it as:
`return check_var('SLE_PRODUCT', 'sled') || (is_sle() && (get_var('FLAVOR') =~ /-Desktop/i));`

I can’t say this version is any more or less readable—it’s a matter of preference without a linter or style guide to dictate the "correct" approach. If you feel strongly about this, I’m happy to adopt a specific convention if there’s one we should adhere to.

Let me know if this works better, or if we’re confident SLE_PRODUCT is sufficient, I can simplify it even further.

Comment (dzedro) [reply lib/version_utils.pm]: If SLE_PRODUCT is not set to SLED then it should if there is no -Desktop flavor
What forces you to use is_sle in function which is testing if the system is SLED ? There is no reason AFAIK.

Comment (dzedro) [reply lib/version_utils.pm]: On SLED `SLE_PRODUCT` should be set to `sled`, done. nothing else is needed.

Comment (volodymyrkatkalov) [reply lib/version_utils.pm]: > On SLED SLE_PRODUCT should be set to sled, done. Nothing else is needed.

Okay, your point is clear now. Then I am reducing code to just `return check_var('SLE_PRODUCT', 'sled')` and expecting variable `SLE_PRODUCT` to be set to `sled` for whatever `sled` oriented tests

Comment (volodymyrkatkalov) [reply lib/version_utils.pm]: P.S.
The phrasing is a bit unclear to me. Could you clarify what you mean by it should be if there is no -Desktop flavor'? Are you suggesting that the logic should explicitly check for FLAVOR not containing -Desktop if SLE_PRODUCT is not set to sled?

Comment (volodymyrkatkalov) [reply lib/version_utils.pm]: @dzedro I think the latest code correction adheres to your suggestion

**Comment by volodymyrkatkalov** [comment]
VRs broken due to https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/8b8c494869d8d9ea942b1dc325209093ef41a15f#diff-21c17412ed1276e21c598100ccd512b81fe3fcd73aa05a02f768d37871e00a3fL11. Need suggestion on alternative schedule yaml

**Comment by dzedro** [comment]
> VRs broken due to [8b8c494#diff-21c17412ed1276e21c598100ccd512b81fe3fcd73aa05a02f768d37871e00a3fL11](https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/8b8c494869d8d9ea942b1dc325209093ef41a15f#diff-21c17412ed1276e21c598100ccd512b81fe3fcd73aa05a02f768d37871e00a3fL11). Need suggestion on alternative schedule yaml

Clarify with @paolostivanin why it was deleted and likely restore that profile or create new one.

**Comment by paolostivanin** [comment]
There was a big FIPS refactoring, now openjdk is included in `fips_*_mode_gnome` where `*` is either `ker` or `env`.

**Comment by volodymyrkatkalov** [comment]
@paolostivanin  I have used `schedule/security/fips/fips_env_mode_gnome.yaml` and have error in `xca`

**Comment by dzedro** [comment]
> @paolostivanin I have used `schedule/security/fips/fips_env_mode_gnome.yaml` and have error in `xca`
xca is not supported/shipped to SLED https://smelt.suse.de/maintained/?q=xca

**Comment by paolostivanin** [comment]
as Jozef wrote, XCA is not supported on SLED. You can use `EXCLUDE_MODULES=xca` to avoid it being scheduled.

**Comment by volodymyrkatkalov** [comment]
@dzedro updated VR with `EXCLUDE_MODULES=xca`

**Review by dzedro — APPROVED** [review]

**Comment by michaelgrifalconi** [comment]
@dzedro I see `is_desktop_installed` imported but not used

**Review by michaelgrifalconi — APPROVED** [review]

**Pull request merged**